torch/custom_class.h

```diff
@@ -153,6 +153,79 @@ class class_ {
     return *this;
   }
 
+  // TODO this is a hack, but it works
+  template <typename GetterFunc, typename SetterFunc = GetterFunc>
+  class_& def_property(
+      std::string name,
+      GetterFunc getterFunc,
+      SetterFunc setterFunc = nullptr,
+      std::string doc_string = "") {
+    torch::jit::Function* getter;
+    torch::jit::Function* setter;
+
+    auto getterName = name + "_getter";
+    auto wrapped_getter =
+        detail::wrap_func<CurClass, GetterFunc>(std::move(getterFunc));
+    getter = process_and_register_property_method(
+        getterName, wrapped_getter, doc_string);
+
+    if (!std::is_same<SetterFunc, GetterFunc>::value) {
```

Comment:
Yeah. The compiler was complaining about nullptr_t. So this is used as a way to figure out if there is any setter method given.

Reply:
I think we talked about this long ago but can you remind us what the issue is with using `nullptr_t` as the default value for `SetterFunc`? I don't think `SetterFunc = GetterFunc` and using that to infer whether there is a getter or setter is a clean solution, even though a getter and setter can't have the same type.

Reply:
This method internally calls this (https://github.com/pytorch/pytorch/blob/22a34bcf4e5eaa348f0117c414c3dd760ec64b13/c10/util/TypeTraits.h#L79) But issue is that compiler errors because it thinks std::nullptr_t doesn't have operator(). Unless our SetterFunc has a default type that defines operator(), I don't think it will work.